### PR TITLE
get transaction list for resource kind

### PR DESCRIPTION
**Pull request by sardhani-ch**
Get transaction list for the resource kind from transaction api, and redirect the user to right screen based on the saved application status

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/controllers/features/common/checkSavedApplicationController.ts [hunk 1]
```diff
@@ -1,30 +1,60 @@
...
     } else {
-        logger.error(`internal server error: ` + savedApplication.status);
+        logger.error("internal server error: " + savedApplication.httpStatusCode);
         const error = new ErrorService();
         error.renderErrorPage(res, locales, lang, BASE_URL + CHECK_SAVED_APPLICATION);
     }
 };
+
+const getRedirectionUrl = async (transactionList: TransactionList|undefined, session: Session, res: Response, locales:any, lang:string): Promise<string> => {
+    const transaction = transactionList?.items[transactionList?.items.length - 1];
```
Comment (sampankumar) [inline src/controllers/features/common/checkSavedApplicationController.ts:37]: The transaction list will be ordered by date in descending order. So take the 1st element.

File: src/controllers/features/common/checkSavedApplicationController.ts [hunk 2]
```diff
@@ -1,30 +1,60 @@
...
     const locales = getLocalesService();
-    if (savedApplication.status === 404) {
-        // Direct to start a new journey if no application found or application was rejected
-        saveDataInSession(req, "resume_application", false);
+
+    if (savedApplication.httpStatusCode === 200) {
+        logger.debug("saved application exists");
+        const transactionlistResource = savedApplication as Resource<TransactionList>;
+        const transactionList = transactionlistResource.resource;
+        const redirectionUrl = getRedirectionUrl(transactionList, session, res, locales, lang);
```
Comment (sampankumar) [inline src/controllers/features/common/checkSavedApplicationController.ts:24]: You can pass the Full transaction Resource to the new function. And also move this function to a service layer please.

File: src/controllers/features/common/checkSavedApplicationController.ts [hunk 3]
```diff
@@ -1,30 +1,60 @@
...
     }
 };
+
+const getRedirectionUrl = async (transactionList: TransactionList|undefined, session: Session, res: Response, locales:any, lang:string): Promise<string> => {
+    const transaction = transactionList?.items[transactionList?.items.length - 1];
+    var url = "";
+    if (transaction?.status !== "closed") {
+        logger.debug("application is open");
+        url = BASE_URL + SAVED_APPLICATION;
+    } else if (transaction.filings![transaction.id + "-1"]?.status === "rejected") {
```
Comment (sampankumar) [inline src/controllers/features/common/checkSavedApplicationController.ts:42]: May be use constants instead of strings in the conditional statements.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/controllers/features/common/checkSavedApplicationController.ts (hunk 1 above)
Comment (sardhani-ch) [reply src/controllers/features/common/checkSavedApplicationController.ts]: resolved

File: src/controllers/features/common/checkSavedApplicationController.ts (hunk 2 above)
Comment (sardhani-ch) [reply src/controllers/features/common/checkSavedApplicationController.ts]: resolved

File: src/controllers/features/common/checkSavedApplicationController.ts (hunk 3 above)
Comment (sardhani-ch) [reply src/controllers/features/common/checkSavedApplicationController.ts]: resolved

**Review by ttingle-ch — APPROVED** [review]
LGTM

**Pull request merged**